logic/state.py

```diff
@@ -1,253 +1,420 @@
+from typing import Optional, Union, NamedTuple
+from numpy.typing import NDArray
 import numpy as np
 import copy
-from api.actions import Move, Push, Idle, Jump, Action
-from util.hexagon import Hex
-from collections import namedtuple
+from util.hexagon import Hexagon, ORIGIN
+from logic.prng import PRNG
+from api.logging import logger as glogger
+from api.actions import ALL_ACTIONS, Action, Idle, Move, Jump, Push
 
 
-RNG = np.random.default_rng()
+# Assert AP is always an integer, this is assumed by the round order tiebreaker
+assert all(isinstance(action.ap, int) for action in ALL_ACTIONS)
 
-Effect = namedtuple('Effect', [
-    'name',
-    'origin',
-    'target',
-    ])
+
+MAX_AP = 100
+REGEN_AP = 50
+NEXT_SEED_ITERATIONS = 100  # Number of iterations on the PRNG to apply between rounds
+CHECK_LEGAL_UNIT_DEPRECATE_DATE = 'September 15, 2022'
```

Comment:
?

Reply:
not sure what it is used for and how but we should stick to a standard date time formatting

Reply:
Yeah I should've dunder'ed this variable name. This is just used to print a warning, it will have to be manually deprecated anyway.